src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp

```diff
@@ -573,16 +369,28 @@ static inline ConvSolution GetSolutionBase(const ConvolutionContext& ctx,
     return result;
 }
 
-ConvSolution ConvAsmImplicitGemmV4R1DynamicFwd::GetSolution(
-    const ConvolutionContext& ctx, const PerformanceImplicitGemmV4R1Dynamic& config, bool) const
+ConvSolution ConvAsmImplicitGemmV4R1DynamicFwd::GetSolution(const ConvolutionContext& ctx) const
 {
-    return GetSolutionBase(ctx, config, AsmImplicitGemmV4R1);
+    auto tunables = GetImplicitGemmV4R1DynamicTunables();
+    auto it       = std::find_if(
+        tunables.begin(), tunables.end(), [&](auto tunable) { return tunable.IsValid(ctx); });
+
+    if(it == tunables.end())
+        MIOPEN_THROW("should not happen!");
+
+    return GetSolutionBase(ctx, *it, AsmImplicitGemmV4R1);
 }
 
-ConvSolution ConvAsmImplicitGemmV4R1DynamicFwd_1x1::GetSolution(
-    const ConvolutionContext& ctx, const PerformanceImplicitGemmV4R1Dynamic& config, bool) const
+ConvSolution ConvAsmImplicitGemmV4R1DynamicFwd_1x1::GetSolution(const ConvolutionContext& ctx) const
 {
-    return GetSolutionBase(ctx, config, AsmImplicitGemmV4R1_1x1);
+    auto tunables = GetImplicitGemmV4R1DynamicTunables();
+    auto it       = std::find_if(
+        tunables.begin(), tunables.end(), [&](auto tunable) { return tunable.IsValid(ctx); });
+
+    if(it == tunables.end())
+        MIOPEN_THROW("should not happen!");
```

Comment:
again

Reply:
OK, let me fix it

Reply:
Please, specify `miopenStatusInternalError` as the error since this is something the user should not observe in the first place.

Reply:
OK, added